iceoryx_posh/include/iceoryx_posh/internal/popo/used_chunk_list.hpp

```diff
@@ -21,161 +22,107 @@
+    /// @note only from runtime context
+    bool insert(mepoo::SharedChunk chunk) noexcept;
+
+    /// @brief Removes a chunk from the list
+    /// @param[in] chunkHeader to look for a corresponding SharedChunk
+    /// @param[out] chunk which is removed
+    /// @return true if successfully removed, otherwise false if e.g. the chunkHeader was not found in the list
+    /// @note only from runtime context
+    bool remove(const mepoo::ChunkHeader* chunkHeader, mepoo::SharedChunk& chunk) noexcept;
+
+    /// @brief Cleans up all the remaining chunks from the list.
+    /// @note from RouDi context once the applications walked the plank
+    void cleanup() noexcept;
 
-            // store chunk mgmt ptr
-            m_data[m_usedListHead] = f_chunk.release();
+  private:
+    void init() noexcept;
 
-            // set freeListHead to the next free entry
-            m_freeListHead = nextFree;
+    bool freeSpaceInList() const noexcept;
 
-            //@todo can we do this cheaper with a global fence in cleanup?
-            m_synchronizer.clear(std::memory_order_release);
-            return true;
-        }
-        else
-        {
-            return false;
-        }
-    }
+  private:
+    static constexpr uint32_t INVALID_INDEX{Capacity};
 
-    // only from runtime context
-    bool remove(const mepoo::ChunkHeader* f_chunkHeader, mepoo::SharedChunk& f_chunk)
+    // this shall be moved to the relative_ptr implementation
+    struct relativ_ptr_data
     {
-        uint32_t previous = InvalidIndex;
-
-        // go through usedList with stored chunks
-        for (uint32_t current = m_usedListHead; current != InvalidIndex; current = m_list[current])
+        constexpr relativ_ptr_data() noexcept
+            : segment(MAX_SEGMENT)
+            , offset(MAX_OFFSET)
         {
-            // does the entry match the one we want to remove?
-            if (m_data[current] != nullptr && m_data[current]->m_chunkHeader == f_chunkHeader)
-            {
-                // return the chunk mgmt entry as SharedChunk object
-                f_chunk = mepoo::SharedChunk(m_data[current]);
-                m_data[current] = nullptr;
-
-                // remove index from used list
-                if (current == m_usedListHead)
-                {
-                    m_usedListHead = m_list[current];
-                }
-                else
-                {
-                    m_list[previous] = m_list[current];
-                }
-
-                // insert index to free list
-                m_list[current] = m_freeListHead;
-                m_freeListHead = current;
-
-                //@todo can we do this cheaper with a global fence in cleanup?
-                m_synchronizer.clear(std::memory_order_release);
-                return true;
-            }
-            previous = current;
+            static_assert(sizeof(relativ_ptr_data) <= 8U, "The relativ_ptr_data size must not exceed 64 bit!");
```

Comment:
oh no, Windows doesn't like this

Reply:
I guess there are some implementation-defined bitfields in the size of structs: https://stackoverflow.com/questions/42314104/clang-and-msvc-report-different-sizeof-for-the-same-struct

I found out that it has on Windows size 16, maybe we do (an ugly) ifdef here or we need to find another way around.